src/parser/mod.rs

```diff
@@ -853,14 +853,30 @@ impl<'a> Parser<'a> {
 
         Ok(Statement::ReleaseSavepoint { name })
     }
-
+    pub fn parse_column_with_data_type(&mut self) -> Result<Expr, ParserError> {
+        self.expect_token(&Token::LBrace)?;
+        let name = self.parse_identifier(false)?;
+        self.expect_token(&Token::Colon)?;
+        let data_type = self.parse_data_type()?;
+        self.expect_token(&Token::RBrace)?;
+        Ok(Expr::ParameterizedViewArg {
+            columns: (name),
+            data_type: (data_type),
+        })
+    }
     /// Parse an expression prefix
     pub fn parse_prefix(&mut self) -> Result<Expr, ParserError> {
         // allow the dialect to override prefix parsing
         if let Some(prefix) = self.dialect.parse_prefix(self) {
             return prefix;
         }
-
+        // ClickHouse support using {column:Datatype} format in its creation of parametrised views
+        match self.peek_token().token {
+            Token::LBrace if dialect_of!(self is ClickHouseDialect) => {
```

Comment:
Can you please also support this syntax in `GenericDialect` as well as explained in https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#new-syntax ?

(also add test for it below)

Reply:
sure